sunpy/map/map_factory.py

```diff
@@ -189,6 +190,11 @@ def _parse_args(self, *args, **kwargs):
 
             arg = args[i]
 
+            # File object
+            if isinstance(arg, io.IOBase):
+                # Extract the name and then rely on the filename logic below
+                arg = arg.name
```

Comment:
If we are doing this, is there really any point?

Reply:
What do you mean by "is there any point"?

Reply:
Well we aren't using the file object, or respecting the context manager. We will be keeping our own file handle open to the fits file once the context manager is closed or once `f.close()` was called, which to me is us outright lying to the user lol

Reply:
Ha, good point... it's going to need a bit more code wrangling then to do it properly.